starter-code/app.js

```diff
@@ -1,50 +1,133 @@
-const express      = require('express');
-const path         = require('path');
-const favicon      = require('serve-favicon');
-const logger       = require('morgan');
+const express = require('express');
+const path = require('path');
+const logger = require('morgan');
 const cookieParser = require('cookie-parser');
-const bodyParser   = require('body-parser');
-const mongoose     = require("mongoose");
-
+const bodyParser = require('body-parser');
+const mongoose = require('mongoose');
+const expressLayouts = require('express-ejs-layouts');
 const app = express();
+const session = require('express-session');
+const MongoStore = require('connect-mongo')(session);
+const bcrypt = require('bcrypt');
+const passport = require('passport');
+const LocalStrategy = require('passport-local').Strategy;
+const flash = require('connect-flash');
+const FbStrategy = require('passport-facebook').Strategy;
+
+const User = require('./models/user');
 
 // Controllers
-const siteController = require("./routes/siteController");
+const index = require('./routes/index');
+const authRoutes = require('./routes/auth-routes');
 
 // Mongoose configuration
-mongoose.connect("mongodb://localhost/ibi-ironhack");
+mongoose.connect('mongodb://localhost/passport-auth', {
+  keepAlive: true,
+  reconnectTries: Number.MAX_VALUE
+});
 
-// view engine setup
+// Middlewares configuration
+app.use(logger('dev'));
+
+// View engine configuration
+app.use(expressLayouts);
 app.set('views', path.join(__dirname, 'views'));
 app.set('view engine', 'ejs');
+app.use(express.static(path.join(__dirname, 'public')));
+app.set('layout', 'layouts/main');
 
-// uncomment after placing your favicon in /public
-//app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
-app.use(logger('combined'));
+// Access POST params with body parser
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
+
+// Authentication / Session
 app.use(cookieParser());
-app.use(express.static(path.join(__dirname, 'public')));
+
+app.use(
+  session({
+    store: new MongoStore({
+      mongooseConnection: mongoose.connection,
+      ttl: 24 * 60 * 60 // 1 day
+    }),
+    secret: 'foobar',
+    resave: true,
+    saveUninitialized: true,
+    cookie: {
+      maxAge: 24 * 60 * 60 * 1000
+    }
+  })
+);
+
+passport.serializeUser((user, cb) => {
```

Comment:
Better put the passport configuration in its own file (inside a folder like helpers, for example).